bring/src/main/java/com/bobocode/petros/bring/context/domain/BeanReference.java

```diff
@@ -0,0 +1,68 @@
+package com.bobocode.petros.bring.context.domain;
+
+import java.util.Objects;
+
+public class BeanReference {
+    private Object beanObject;
+    private boolean singleton = true;
+    private boolean prototype = false;
```

Comment:
maybe better to create one scope Enum field instead of these two boolean flags

Reply:
> maybe better to create one scope Enum field instead of these two boolean flags?

We talked to @vromantsev and decided to create two fields because it's weird to call the isProtoripe method to check if the field is a singleton.

Reply:
> > maybe better to create one scope Enum field instead of these two boolean flags?
> 
> We talked to @vromantsev and decided to create two fields because it's weird to call the isProtoripe method to check if the field is a singleton.

then you need to invoke two setters to change the scope

Reply:
If we are talking about invoking 1 or 2 setters - to be honest it's not a problem, except in case you meant something else behind the setter invocation

Reply:
> > > maybe better to create one scope Enum field instead of these two boolean flags?
> > 
> > 
> > We talked to @vromantsev and decided to create two fields because it's weird to call the isProtoripe method to check if the field is a singleton.
> 
> then you need to invoke two setters to change the scope

We can also use emuns

Reply:
But I'm not sure that scope field should be final, because if so, we won't be able to declare it as prototype due to final modifier limitation.